### ISSUE TITLE
[Accessibility], 2.4.4. WCAG Term selection in Vocabularies is not accessible for screen readers

### ISSUE THREAD

**Issue by Eiskore** [NatLibFi/Skosmos#1980]
## URL address of the page where you encountered the problem
https://test.dev.finto.fi/yso/en/ and other vocabulary pages

## Description of the problem
Screen readers don't read the terms in the sidebar concept list correctly. For each link in the listing, aria-label="Mene käsitesivulle" has been added as a kind of supplementary text. However, the aria-label overrides the other content of the link, so screen readers only read the text “Mene käsitesivulle”/"Go to concept page" but not the visible name, e.g. "aakkostus", "AA liike", etc. The link list is therefore currently unusable for screen readers.
The problem is repeated in all tabs and in all the vocabularies studied.
This is a blocker issue for screen reader users.

**Fix**: Aria-label should not be used to add only additional information. Aria-label can be used to add the name of the link/ treeitem in its entirety (e.g. aria-label="alphabet, open concept"). However, if you only want to add additional information after the term, the recommended way to do it is to add an invisible additional information text, which can be found with screen readers https://www.w3.org/WAI/WCAG21/Techniques/css/C7 

**How to test:** Turn on  NVDA. Navigate to vocabulary tab panel. Open some tab, that contains links or treeitems. Navigate to tab panel and browse the links or treeitems:

Expected result: NVDA reads the names of the links
Actual result: NVDA reads “Go to the concept page / Mene käsitesivulle”

Severity: Critical
Priority: High

## Additional information (e.g. screenshots) about the problem

<img width="1004" height="573" alt="Image" src="https://github.com/user-attachments/assets/e8954331-23ef-49f3-9555-7e460f203c0a" />

<img width="1004" height="332" alt="Image" src="https://github.com/user-attachments/assets/e870c197-3816-424d-8d6b-b1d3d09dbb71" />

## The browser you used when the problem appeared
Chrome, Firefox, Edge

**Comment by tinskuu** [comment]
This is fixed, links have unique names.
